pocld/daemon.cc

```diff
@@ -574,8 +619,9 @@ void PoclDaemon::readAllClientSocketsThread() {
             IncompleteRequests.push_back(new Request());
             FdsChanged = true;
             /* XXX: Set these based on CreateOrAttachSession request instead? */
-            pocl_remote_client_set_socket_options(newfd, Params.BufSize,
-                                                  Params.IsFast);
+            if (!UseVsock) /* vsock don't support TCP_NODELAY */
+              pocl_remote_client_set_socket_options(newfd, Params.BufSize,
```

Comment:
Same here, if any of the other socket options make sense for vsock, I'd rather adjust this function to cope with vsock and call it for all socket types.